tests/unit/test_client.py

```diff
@@ -509,18 +521,27 @@ def test_client_signs_call(self, signer_mock):
             'test_operation', request)
 
     def test_client_makes_call_with_error(self):
-        error_response = {
-            'Error': {'Code': 'code', 'Message': 'error occurred'}
-        }
-        self.endpoint.make_request.return_value = (
-            mock.Mock(status_code=400), error_response)
+        error_responses = [
+            {'Error': {'Code': 'code', 'Message': 'error occurred'}},
+            {'Error': {'Code': '404', 'Message': 'Not Found'}},
+            {'Error': {'Code': 'InvalidAddress.NotFound', 'Message': ''}},
+            {'Error': {'Code': 'TestOperationException', 'Message': ''}},
+            {'Error': {'Message': 'error occurred'}},
+            {'Error': {}}
```

Comment:
How do cases when the ``Code`` is an empty string get handled? It may be good to have a test specifying that behavior.